src/java/generator/MapGenerator.java

```diff
@@ -834,31 +835,37 @@ private void setupTerrainPipeline() {
-        ConcurrentFloatMask heightmapCliffs = new ConcurrentFloatMask(mapSize + 1, random.nextLong(), symmetrySettings, "heightmapCliffs");
-        ConcurrentFloatMask heightmapShore = new ConcurrentFloatMask(mapSize + 1, random.nextLong(), symmetrySettings, "heightmapShore");
-        ConcurrentFloatMask heightmapHills = new ConcurrentFloatMask(mapSize + 1, random.nextLong(), symmetrySettings, "heightmapHills");
-        ConcurrentFloatMask heightmapValleys = new ConcurrentFloatMask(mapSize + 1, random.nextLong(), symmetrySettings, "heightmapValleys");
-        ConcurrentBinaryMask oceanFloor = land.copy().invert();
-
-        oceanFloor.acid(.01f, 1).erode(.75f, symmetrySettings.getSpawnSymmetry(), 2).smooth(32, .75f).invert().removeAreasSmallerThan(128);
-
-        heightmapBase.init(land, waterHeight + .5f, waterHeight + .5f);
-        heightmapPlateaus.init(plateaus, 0, PLATEAU_HEIGHT).smooth(10, ramps);
-        heightmapHills.init(hills, 0, HILL_HEIGHT).smooth(16, land.copy().minus(plateaus));
-        heightmapValleys.init(valleys, 0, VALLEY_HEIGHT).smooth(16, plateaus);
-        heightmapLand.add(heightmapHills).add(heightmapValleys).maskToOceanHeights(0.35f, land).clampMin(biome.getWaterSettings().getElevationAbyss() - waterHeight + 1f).maskToOceanHeights(0.15f, oceanFloor).clampMin(biome.getWaterSettings().getElevationAbyss() - waterHeight - 1f);
-        heightmapCliffs.init(cliffs, 0, 1f).maskToMountains(cliffs);
-        heightmapShore.init(shore, 0, 1.5f).maskToMountains(shore);
-        heightmapMountains.maskToMountains(mountains).add(heightmapCliffs).add(heightmapShore);
-
-        ConcurrentBinaryMask mountainsPresent = new ConcurrentBinaryMask(heightmapMountains, 3f, random.nextLong(), "mountainsPresent");
-
-        heightmapMountains.add(mountainsPresent.copy().minus(plateaus), PLATEAU_HEIGHT).subtract(mountainsPresent.copy().invert(), 10f).clampMin(0);
-
-        heightmapBase.add(heightmapMountains).add(heightmapLand).add(heightmapPlateaus).smooth(1);
+        ConcurrentBinaryMask land2 = land.copy().setSize(mapSize + 1);
+
+        List<String> allBrushes = Arrays.asList("buttons.png", "crescent.png", "crystal.png", "hawaii1.png", "hawaii2.png", "hill1.png", "hill2.png", "mountain1.png", "mountain2.png", "mountain3.png", "mountain4.png", "mountain5.png", "mountain6.png", "mountain7.png", "mountain8.png", "mountain9.png", "noise1.png", "noise2.png", "striations.png", "structured.png", "sun.png", "volcano2.png");
+        List<String> okayBrushes = Arrays.asList("hawaii1.png", "hawaii2.png", "hill1.png", "hill2.png", "mountain1.png", "mountain2.png", "mountain3.png", "mountain4.png", "mountain5.png", "mountain6.png", "mountain7.png", "mountain8.png", "structured.png", "volcano2.png");
+        List<String> goodBrushes = Arrays.asList("hawaii2.png", "hill1.png", "mountain1.png", "mountain2.png", "mountain4.png", "mountain5.png", "mountain6.png", "mountain7.png", "volcano2.png");
```

Comment:
These should probably be static lists in the Brushes class and not defined in the generator class. Also if only goodBrushes is used better to only have that one defined.

Reply:
moved